lls/main.c

```diff
@@ -170,18 +174,43 @@ lls_proc(void *arg)
 	struct net_config *net_conf = get_net_conf();
 	uint32_t lcore = rte_lcore_id();
 	uint16_t queue_id = get_queue_id();
+	bool lacp_enabled = false;
+	uint32_t source_ip;
+	uint8_t source_port;
+
+	if (net_conf->front.lacp_enabled) {
+		lacp_enabled = true;
+		source_ip = net_conf->front.ipv4_addr;
+		source_port = net_conf->front.id;
+	} else if (net_conf->back.lacp_enabled) {
+		lacp_enabled = true;
+		source_ip = net_conf->back.ipv4_addr;
+		source_port = net_conf->back.id;
+	}
 
 	RTE_LOG(NOTICE, GATEKEEPER,
 		"lls: the LLS block is running at lcore = %u\n", lcore);
 
+	/*
+	 * NOTE: For LACP to work within the expected
+	 * latency, rte_eth_rx_burst()/rte_eth_tx_burst()
+	 * should be called every 100 ms, so this
+	 * while loop should loop at least every 100 ms.
+	 *
+	 * XXX Could add some calls to gettimeofday() here
+	 * to measure how quick the while loop is looping.
+	 */
 	while (likely(!exiting)) {
-
 		/* Read in packets on front and back interfaces. */
 		process_pkts(&net_conf->front, queue_id);
 		process_pkts(&net_conf->back, queue_id);
 
 		/* Process any ARP updates. */
 		arp_cache_writer();
+
+		/* Send gratuitous ARP to fulfill LACP requirement. */
```

Comment:
If LACP is enabled on both interfaces, don't you need to transmit a packet on the second interface as well?